server/blob/blob-s3/src/main/java/org/apache/james/blob/objectstorage/aws/S3BlobStoreDAO.java

```diff
@@ -154,23 +157,24 @@ public Publisher<InputStream> readReactive(BucketName bucketName, BlobId blobId)
             .map(res -> ReactorUtils.toInputStream(res.flux));
     }
 
-    @Override
-    public Publisher<ReactiveByteSource> readAsByteSource(BucketName bucketName, BlobId blobId) {
-        BucketName resolvedBucketName = bucketNameResolver.resolve(bucketName);
-
-        return getObject(resolvedBucketName, blobId)
-            .onErrorMap(NoSuchBucketException.class, e -> new ObjectNotFoundException("Bucket not found " + resolvedBucketName.asString(), e))
-            .onErrorMap(NoSuchKeyException.class, e -> new ObjectNotFoundException("Blob not found " + blobId.asString() + " in bucket " + resolvedBucketName.asString(), e))
-            .map(res -> new ReactiveByteSource(res.sdkResponse.contentLength(), res.flux));
-    }
-
     private static class FluxResponse {
         final CompletableFuture<FluxResponse> supportingCompletableFuture = new CompletableFuture<>();
         GetObjectResponse sdkResponse;
         Flux<ByteBuffer> flux;
     }
 
     private Mono<FluxResponse> getObject(BucketName bucketName, BlobId blobId) {
+        return getObjectFromStore(bucketName, blobId)
+            .onErrorResume(e -> {
```

Comment:
Please do this only if it is not found IMO 

No need to retry arbitrary errors.